### PR TITLE
Fix can't access IPV6 address via management interface because 'default' route table does not add to route lookup issue.

### PR DESCRIPTION
**Pull request by liuh-80**
Fix can't access IPV6 address via management interface because 'default' route table does not add to route lookup issue.

#### Why I did it
When device set with IPV6 TACACS server address, and shutdown all BGP, device can't connect to TACACS server via management interface.

After investigation, I found the IPV6 'default' route table does not add to route lookup:

admin@vlab-01:~$ ip -6 rule list
1001:   from all lookup local
32765:  from fec0::ffff:afa:1 lookup default
32766:  from all lookup main
admin@vlab-01:~$

As compare:
admin@vlab-01:~$ ip -4 rule list
1001:   from all lookup local
32764:  from all to 172.17.0.1/24 lookup default
32765:  from 10.250.0.101 lookup default
32766:  from all lookup main
32767:  from all lookup default <== 'default' route table exist in IPV4 route lookup

Issue fix by add 'default' route table to route lookup with following command:
admin@vlab-01:~$ sudo ip -6 rule add pref 32767 lookup default
admin@vlab-01:~$ ip -6 rule list
1001:   from all lookup local
32765:  from fec0::ffff:afa:1 lookup default
32766:  from all lookup main
32767:  from all lookup default <== 'default' route table been added to IPV6 route lookup
admin@vlab-01:~$

##### Work item tracking
- Microsoft ADO: 25798732

#### How I did it
When management interface using 'default' route table, add 'default' route table to IPV6 route lookup.

#### How to verify it
Pass all UT.
Add new UT to cover this change.
Manually verify issue fixed:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [] 202205
- [] 202211
- [] 202305

#### Tested branch (Please provide the tested image version)

- [x]  master-17281.417570-2133d58fa

#### Description for the changelog
Fix can't access IPV6 address via management interface because 'default' route table does not add to route lookup issue.

### CODE REVIEW

**Comment by liuh-80** [comment]
Created UT for this change: https://github.com/sonic-net/sonic-mgmt/pull/10842

File: files/image_config/interfaces/interfaces.j2 [hunk 1]
```diff
@@ -94,6 +94,11 @@ iface {{ name }} {{ 'inet' if prefix | ipv4 else 'inet6' }} static
 {% for route in MGMT_INTERFACE[(name, prefix)]['forced_mgmt_routes'] %}
     pre-down ip rule delete pref 32764 to {{ route }} table {{ vrf_table }}
 {% endfor %}
+{% if prefix | ipv6 and vrf_table == 'default'%}
+    # IPV6 default table not add to lookup by default, management server need this to access IPV6 address when BGP shutdown
+    up ip -6 rule add pref 32767 lookup {{ vrf_table }}
```
Comment (qiluo-msft) [inline files/image_config/interfaces/interfaces.j2:99]: >32767 [](http://example.com/codeflow?start=27&length=5)

How do you get the magic number?

**Comment by qiluo-msft** [comment]
@prsunny Could you review this PR?

File: files/image_config/interfaces/interfaces.j2 (hunk 1 above)
Comment (liuh-80) [reply files/image_config/interfaces/interfaces.j2:99]: The number is based on IPV4 default route table priority:

admin@vlab-01:~$ ip -4 rule list
1001: from all lookup local
32764: from all to 172.17.0.1/24 lookup default
32765: from 10.250.0.101 lookup default
32766: from all lookup main
**32767**: from all lookup default

The default route table need have lowest priority so when can' find route from 'local' and 'main' table the rules from 'default' table will be use.

File: src/sonic-config-engine/tests/sample_output/py2/interfaces [hunk 2]
```diff
@@ -42,6 +42,9 @@ iface eth0 inet6 static
     pre-down ip -6 route delete default via 2603:10e2:0:2902::1 dev eth0 table default
     pre-down ip -6 route delete 2603:10e2:0:2902::/64 dev eth0 table default
     pre-down ip -6 rule delete pref 32765 from 2603:10e2:0:2902::8/128 table default
+    # IPV6 default table not add to lookup by default, management server need this to access IPV6 address when BGP shutdown
+    up ip -6 rule add pref 32767 lookup default
```
Comment (prsunny) [inline src/sonic-config-engine/tests/sample_output/py2/interfaces:46]: can you please move this to 'up' section - line 40

File: src/sonic-config-engine/tests/sample_output/py2/two_mgmt_interfaces [hunk 3]
```diff
@@ -57,6 +57,9 @@ iface eth1 inet6 static
     pre-down ip -6 route delete default via 2603:10e2:0:abcd::1 dev eth1 table default
     pre-down ip -6 route delete 2603:10e2:0:abcd::/64 dev eth1 table default
     pre-down ip -6 rule delete pref 32765 from 2603:10e2:0:abcd::8/128 table default
+    # IPV6 default table not add to lookup by default, management server need this to access IPV6 address when BGP shutdown
+    up ip -6 rule add pref 32767 lookup default
```
Comment (prsunny) [inline src/sonic-config-engine/tests/sample_output/py2/two_mgmt_interfaces:61]: same here. please keep 'up' and 'pre-down' separate

File: files/image_config/interfaces/interfaces.j2 (hunk 1 above)
Comment (qiluo-msft) [reply files/image_config/interfaces/interfaces.j2:99]: Seems we need to define a named constant for 32764, and +3 here.

File: src/sonic-config-engine/tests/sample_output/py2/interfaces (hunk 2 above)
Comment (liuh-80) [reply src/sonic-config-engine/tests/sample_output/py2/interfaces]: Fixed

**Review by prsunny — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by StormLiangMS** [comment]
hi @liuh-80  could you help to run test on 202305 image with this PR? We'd like to avoid regression at this stage.

**Comment by liuh-80** [comment]
> ip -6 rule list



> hi @liuh-80 could you help to run test on 202305 image with this PR? We'd like to avoid regression at this stage.

@StormLiangMS , I verified with SONiC.202305-17658.446964-c9b5bb179, after cherry-pick, the issue fixed:

admin@vlab-01:~$ ip -6 rule list
1001:   from all lookup local
32765:  from fec0::ffff:afa:1 lookup default
32766:  from all lookup main
32767:  from all lookup default <== IPV6 default table added

**Pull request merged**